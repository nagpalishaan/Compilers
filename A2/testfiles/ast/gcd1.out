Program
  List
    Function
      IdDecl ID="gcd1"
      Opt
        List
          Param
            IdDecl ID="a"
          Param
            IdDecl ID="b"
      Opt
        List
          While
            Not
              IdUse ID="a"
              IdUse ID="b"
            List
              IfElse
                Greater
                  IdUse ID="a"
                  IdUse ID="b"
                List
                  AssignVar
                    IdDecl ID="a"
                    Sub
                      IdUse ID="a"
                      IdUse ID="b"
                Opt
                  List
                    AssignVar
                      IdDecl ID="b"
                      Sub
                        IdUse ID="b"
                        IdUse ID="a"
          Return
            IdUse ID="a"
