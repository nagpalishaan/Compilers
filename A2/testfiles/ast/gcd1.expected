Program
  List
    Function
      IdDecl ID="gcd1"
      Opt
        List
          Param
            IdDecl ID="a"
          Param
            IdDecl ID="b"
      List
        While
          Not
            IdUse ID="a"
            IdUse ID="b"
          List
            IfElse
              Greater
                IdUse ID="a"
                IdUse ID="b"
              List
                AssignVar
                  IdUse ID="a"
                  Sub
                    IdUse ID="a"
                    IdUse ID="b"
              Opt
                List
                  AssignVar
                    IdUse ID="b"
                    Sub
                      IdUse ID="b"
                      IdUse ID="a"
        Return
          IdUse ID="a"
